src/sentry/eventstream/kafka/backend.py

```diff
@@ -210,8 +216,16 @@ def handle_shutdown_request(signum: int, frame: Any) -> None:
             i = i + 1
             owned_partition_offsets[key] = message.offset() + 1
 
-            with metrics.timer("eventstream.duration", instance="get_task_kwargs_for_message"):
-                task_kwargs = get_task_kwargs_for_message(message.value())
+            use_kafka_headers = options.get("post-process-forwarder:kafka-headers")
+
+            if use_kafka_headers is True:
+                with metrics.timer(
+                    "eventstream.duration", instance="get_task_kwargs_for_message_from_headers"
+                ):
+                    task_kwargs = get_task_kwargs_for_message_from_headers(message.headers())
```

Comment:
If this call fails, couldn't we call the old function? It also doesn't seem hard to check if the headers exist, and if they're missing use the old version. That way we don't have to wait for older messages that are missing the headers to be cleared before turning the flag on.

Reply:
I added the try/catch around the whole dispatch behavior. I don't think we should attempt to be clever around detecting whether the headers exist, let's just respect the option picked. Practically, the wait is probably like a number of seconds after deployment that we can't turn it on and we are unlikely to attempt that.